pytket/extensions/quantinuum/backends/quantinuum.py

```diff
@@ -1510,11 +1510,42 @@ def _convert_result(
     n_bits: Optional[int] = None,
     results_selection: Optional[list[tuple[str, int]]] = None,
 ) -> BackendResult:
+
+    for creg, reslist in resultdict.items():
+        if any(["-" in res for res in reslist]):
+            raise ValueError(
+                f"found negative value for creg: {creg}. \
+This could indicate a problem with the circuit submitted"
+            )
+
     if results_selection is None:
-        array_dict = {
-            creg: np.array([list(a) for a in reslist]).astype(np.uint8)
-            for creg, reslist in resultdict.items()
-        }
+        found_int_res = False
+        for creg, reslist in resultdict.items():
+            if any([re.findall("[23456789]", res) for res in reslist]):
+                found_int_res = True
+
+        if found_int_res:
+            # this is only a temporary solution and not fully working
```

Comment:
This isn't reliable; we could still miss integer results in some cases (e.g. "10").
Is there some way we can know what to expect when we submit the circuit, and store this information in the `ResultHandle`, instead of guessing from the results?

Do you know when we can expect a fix for this in L4?

Reply:
I agree, that is why I labeled this as a `temporary solution`. We could parse the QIR given by the user and try to predict it? We don't have the circuit, so we can't do any predictions based on the number of bits.
I am not sure when the fix is going to happen.